element_array_ephys/ephys_no_curation.py

```diff
@@ -1644,99 +1765,6 @@ def get_openephys_probe_data(ephys_recording_key: dict) -> list:
     return probe_data
 
 
-def get_neuropixels_channel2electrode_map(
-    ephys_recording_key: dict, acq_software: str
-) -> dict:
-    """Get the channel map for neuropixels probe."""
-    if acq_software == "SpikeGLX":
-        spikeglx_meta_filepath = get_spikeglx_meta_filepath(ephys_recording_key)
-        spikeglx_meta = spikeglx.SpikeGLXMeta(spikeglx_meta_filepath)
-        electrode_config_key = (
-            EphysRecording * probe.ElectrodeConfig & ephys_recording_key
-        ).fetch1("KEY")
-
-        electrode_query = (
-            probe.ProbeType.Electrode * probe.ElectrodeConfig.Electrode
-            & electrode_config_key
-        )
-
-        probe_electrodes = {
-            (shank, shank_col, shank_row): key
-            for key, shank, shank_col, shank_row in zip(
-                *electrode_query.fetch("KEY", "shank", "shank_col", "shank_row")
-            )
-        }
-
-        channel2electrode_map = {
-            recorded_site: probe_electrodes[(shank, shank_col, shank_row)]
-            for recorded_site, (shank, shank_col, shank_row, _) in enumerate(
-                spikeglx_meta.shankmap["data"]
-            )
-        }
-    elif acq_software == "Open Ephys":
-        probe_dataset = get_openephys_probe_data(ephys_recording_key)
-
-        electrode_query = (
-            probe.ProbeType.Electrode * probe.ElectrodeConfig.Electrode * EphysRecording
-            & ephys_recording_key
-        )
-
-        probe_electrodes = {
-            key["electrode"]: key for key in electrode_query.fetch("KEY")
-        }
-
-        channel2electrode_map = {
-            channel_idx: probe_electrodes[channel_idx]
-            for channel_idx in probe_dataset.ap_meta["channels_indices"]
-        }
-
-    return channel2electrode_map
-
-
-def generate_electrode_config(probe_type: str, electrode_keys: list) -> dict:
```

Comment:
I think it makes sense to keep this function, as it is reusable in may places

Reply:
@ttngu207. I've made this [commit](https://github.com/datajoint/element-array-ephys/pull/179/commits/1d805849f20ee1e5ce6a911d06d189f9d699900f) to address your comment. It's true that there's some code duplication but I think this enhances separation of concern (compute vs. insert) and readability. Previously it wasn't very clear what that function does and where probe.ElectrodeConfig was being populated. I'm open to any suggestions though.

Reply:
Right, it wasn't obvious where/how entries for `probe.ElectrodeConfig` were generated/inserted.
I agree with this new change. I'd suggest to go a bit further and update the `generate_electrode_config_name` function to be `generate_electrode_config_entry` - returning `(econfig_entry, econfig_electrodes)` (including the hash)

Then you just use that for the insert, making the insert more visible without duplicating a lot of code

```
if not probe.ElectrodeConfig & {"electrode_config_hash": econfig_entry['electrode_config_hash']}:
        probe.ElectrodeConfig.insert1(econfig_entry)
        probe.ElectrodeConfig.Electrode.insert(econfig_electrodes)
 ```
 (something like that)

Reply:
@ttngu207 Sounds good. I've updated the function.